button-card.js

```diff
@@ -121,118 +127,173 @@
       <ha-card style="color: ${fontColor};" @tap="${ev => this._toggle(state, config)}">
-        <button-card-button style="background-color: ${color}; ${config.card_style}">
-        <div style="${config.card_style}">
-          ${config.icon || icon ? html`<ha-icon style="width: ${config.size}; height: ${config.size};" icon="${icon}"></ha-icon>` : ''}
-          ${config.name ? html`<span>${config.name}</span>` : ''}
-          ${config.show_state ? html`<span>${state.state} ${state.attributes.unit_of_measurement ? state.attributes.unit_of_measurement : ''}</span>` : ''}
+        <button-card-button class="${this.isClickable(state, config) ? '' : "disabled"}" style="background-color: ${color}; ${config.card_style}">
+        <div class="main" style="${style}">
+          ${icon ? html`<ha-icon style="width: ${config.size}; height: ${config.size};" icon="${icon}"></ha-icon>` : ''}
+          ${config.name ? html`<div>${config.name}</div>` : ''}
+          ${config.show_state ? html`<div>${state.state} ${state.attributes.unit_of_measurement ? state.attributes.unit_of_measurement : ''}</div>` : ''}
          </div>
         </button-card-button>
       </ha-card>
       `;
     }
 
-    iconColoredHtml(state, config) {
-      const color = this.buildCssColorAttribute(state, config);
-      const icon = this.buildIcon(state, config);
+    iconColoredHtml(state, config, configState) {
+      const color = this.buildCssColorAttribute(state, config, configState);
+      const icon = this.buildIcon(state, config, configState);
+      const style = this.buildStyle(state, config, configState);
       return html`
       <ha-card @tap="${ev => this._toggle(state, config)}">
-        <button-card-button style="${config.card_style}">
-        <div style="${config.card_style}">
-          ${config.icon || icon ? html`<ha-icon style="color: ${color}; width: ${config.size}; height: ${config.size};" icon="${icon}"></ha-icon>` : ''}
-          ${config.name ? html`<div>${config.name}</div>` : ''}
-          ${config.show_state ? html`<div>${state.state} ${state.attributes.unit_of_measurement ? state.attributes.unit_of_measurement : ''}</div>` : ''}
-        </div>
+        <button-card-button class="${this.isClickable(state, config) ? '' : "disabled"}" style="${config.card_style}">
+          <div class="main" style="${style}">
+            ${icon ? html`<ha-icon style="color: ${color}; width: ${config.size}; height: ${config.size};" icon="${icon}"></ha-icon>` : ''}
+            ${config.name ? html`<div>${config.name}</div>` : ''}
+            ${config.show_state ? html`<div>${state.state} ${state.attributes.unit_of_measurement ? state.attributes.unit_of_measurement : ''}</div>` : ''}
+          </div>
         </button-card-button>
       </ha-card>
-      `;
+        `;
     }
 
     setConfig(config) {
       // if (!config.entity) {
       //   throw new Error('You need to define entity');
       // }
       this.config = { ...config };
```

Comment:
This should allow us to clean up `setConfig` and set the default tap_action's action. 

```suggestion
      this.config = {
          tap_action: {action: "toggle"},
          size: '40%',
          color_type: 'icon',
          color_off: 'var(--disabled-text-color)'
          default_color: 'var(--primary-text-color)'
          ...config
     };
```